src/routes/snippets.rs

```diff
@@ -155,15 +155,15 @@ fn split_marker(mut snippets: Vec<Snippet>, limit: usize) -> (Option<String>, Ve
 
 #[allow(clippy::too_many_arguments)]
 #[get("/snippets?<title>&<syntax>&<tag>&<marker>&<limit>")]
-pub fn list_snippets<'h>(
-    storage: State<Box<dyn Storage>>,
-    origin: &Origin,
+pub async fn list_snippets<'h>(
+    storage: &State<Box<dyn Storage>>,
+    origin: &Origin<'_>,
     title: Option<String>,
     syntax: Option<String>,
     tag: Option<String>,
-    limit: Option<Result<PaginationLimit, ApiError>>,
+    limit: Result<PaginationLimit, rocket::form::Errors<'_>>,
```

Comment:
How come it's not an optional anymore? Is this because `limit` is always somewhere set? Is this because of `Default` trait?

Reply:
Good question!

* The form API changed, and we can no longer return our own `ApiError` from there directly
* somehow `Option` messes up route matching, and Rocket does not call `list_snippets()` anymore

But also, yes, because we also implement the `Default` trait, `Option` + `Result` are redundant, and we only need the latter.